pkg/proc/core/linux_core.go

```diff
@@ -155,8 +224,8 @@ func readNotes(core *elf.File) ([]*Note, error) {
 	return notes, nil
 }
 
-// readNote reads a single note from r, decoding the descriptor if possible.
-func readNote(r io.ReadSeeker) (*Note, error) {
+// readNoteAMD64 reads a single note from r, decoding the descriptor if possible.
+func readNoteAMD64(r io.ReadSeeker) (*Note, error) {
```

Comment:
This function and `readNoteARM64` are basically the same. Rename this to `readNote`, remove `readNoteARM64`. Add a `machineType` parameter to this. Use the `machineType` parameter to decide what to do inside the switch cases `elf.NT_PRSTATUS`, `elf.NT_X86_STATE` (return an error if machineType is not amd64) and `elf.NT_FPREGSET`.